src/app/base/components/node/StorageTables/AvailableStorageTable/AddPartition/AddPartition.test.tsx

```diff
@@ -149,69 +107,63 @@ describe("AddPartition", () => {
         }),
       }),
     });
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter>
-          <CompatRouter>
-            <AddPartition
-              closeExpanded={jest.fn()}
-              disk={disk}
-              systemId="abc123"
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <AddPartition closeExpanded={jest.fn()} disk={disk} systemId="abc123" />,
+      { route: "/", state }
     );
 
     // Set logical volume size to 2GB
-    await act(async () => {
-      wrapper.find("input[name='partitionSize']").simulate("change", {
-        target: { name: "partitionSize", value: "2" },
-      } as React.ChangeEvent<HTMLInputElement>);
+    const partitionSize = screen.getByRole("spinbutton", {
+      name: /Size/i,
     });
-    wrapper.update();
-
+    await userEvent.clear(partitionSize);
+    await userEvent.type(partitionSize, "2");
     expect(
-      wrapper
-        .find(".p-form-validation__message")
-        .text()
-        .includes("available in this disk")
-    ).toBe(true);
+      screen.getByText(/Only 1GB available in this disk/i)
+    ).toBeInTheDocument();
   });
 
-  it("correctly dispatches an action to create a partition", () => {
+  it("correctly dispatches an action to create a partition", async () => {
     const disk = diskFactory({ id: 1 });
     const state = rootStateFactory({
       machine: machineStateFactory({
-        items: [machineDetailsFactory({ disks: [disk], system_id: "abc123" })],
+        items: [
+          machineDetailsFactory({
+            disks: [disk],
+            system_id: "abc123",
+            supported_filesystems: [{ key: "fat32", ui: "FAT32" }],
+          }),
+        ],
         statuses: machineStatusesFactory({
           abc123: machineStatusFactory(),
         }),
       }),
     });
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter>
-          <CompatRouter>
-            <AddPartition
-              closeExpanded={jest.fn()}
-              disk={disk}
-              systemId="abc123"
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <AddPartition closeExpanded={jest.fn()} disk={disk} systemId="abc123" />,
+      { route: "/", store }
```

Comment:
Did this test (and other in this file) fail if you didn't pass anything as the route?